app/src/main/java/com/es/findsoccerplayers/AccountActivity.java

```diff
@@ -0,0 +1,96 @@
+package com.es.findsoccerplayers;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class AccountActivity extends AppCompatActivity {
+
+    private Button btnLogOut;
+    private FirebaseAuth mAuth;
+    private FirebaseUser acct;
+    private GoogleSignInClient mGoogleSignInClient;
+    private TextView id;
+    private TextView email;
+    private TextView nome;
+    private ImageView photo;
+    private Toolbar toolbar;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_account);
+
+        btnLogOut=findViewById(R.id.btnLogOut);
+        mAuth=FirebaseAuth.getInstance();
+        acct = FirebaseAuth.getInstance().getCurrentUser();
+        id = (TextView) findViewById(R.id.id);
+        email = (TextView) findViewById(R.id.email);
+        nome = (TextView) findViewById(R.id.nome);
+        toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        btnLogOut.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                signOut();
+                Intent intent=new Intent(AccountActivity.this, MainActivity.class);
+                startActivity(intent);
+            }
+        });
+
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
```

Comment:
ma questo è codice da riciclare (nel senso c'è un frammento simile in un altro file). Perché non fare una classe che wrappa la GoogleSignIn?

Reply:
beh si può fare, ma si tratta di 5 righe

Reply:
si alla fine sono poche righe, è uguale